src/App.tsx

```diff
@@ -1,17 +1,66 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback, useContext, useEffect, useRef, useState,
+} from 'react';
+import { ErrorOfTodo } from './components/ErrorOfTodo';
+import { ListOfTodo } from './components/ListOfTodo';
+import { NewTodo } from './components/NewTodo';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [userTodos, setUserTodos] = useState<Todo[]>([]);
+  const [hasTodos, setHasTodos] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [query, setQuery] = useState('');
+  const [isEditing, setIsEditing] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleLoadTodos = useCallback(async () => {
+    try {
+      if (user) {
+        const todosFromServer = await getTodos(user.id);
+
+        setUserTodos(todosFromServer);
+        setIsEditing(false);
+      }
+    } catch (err) {
+      setHasError(true);
+      setErrorMessage('Server error!');
+
+      setTimeout(() => {
+        setHasError(false);
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, []);
+
+  const handleErrorClose = useCallback(() => setHasError(false), []);
+
+  useEffect(() => {
+    // focus the element with `ref={newTodoField}`
+    handleLoadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (userTodos.length !== 0) {
+      setHasTodos(true);
+    } else {
+      setHasTodos(false);
+    }
+  }, [userTodos]);
```

Comment:
Why do we need this useEffect and state? We can just compare length to 0 and save this result to constant